src/pint/models/timing_model.py

```diff
@@ -361,6 +361,8 @@ def orbital_phase(self, barytimes, anom="mean", radians=True):
                 bts = np.asarray(barytimes.mjd)
             else:
                 raise ValueError("barytimes as Time instance needs scale=='tdb'")
+        elif isinstance(barytimes, MJDParameter):
+            bts = np.asarray(barytimes.value)
```

Comment:
Likewise, `.value` versus `.quantity.to_value(u.day)`?

Reply:
That also doesn't work.  Maybe our MJDParameter needs work:
```
In [8]: mm.TASC.quantity.to_value(u.day)                                                                 
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-8-a2d9b63ca624> in <module>
----> 1 mm.TASC.quantity.to_value(u.day)

~/python_venvs/py3/lib/python3.8/site-packages/astropy/time/core.py in to_value(self, format, subfmt)
    983         # only, to make life easier for TimeDelta.to_value()).
    984         if format not in self.FORMATS:
--> 985             raise ValueError('format must be one of {}'
    986                              .format(list(self.FORMATS)))
    987 

ValueError: format must be one of ['jd', 'mjd', 'decimalyear', 'unix', 'unix_tai', 'cxcsec', 'gps', 'plot_date', 'stardate', 'datetime', 'ymdhms', 'iso', 'isot', 'yday', 'datetime64', 'fits', 'byear', 'jyear', 'byear_str', 'jyear_str', 'pulsar_mjd', 'mjd_long', 'pulsar_mjd_long', 'mjd_string', 'pulsar_mjd_string']
```

Reply:
Maybe `MJDParameter` should look like a Quantity with units of days? possibly with an extra property to get the result as a Time object? By the time it's in a Parameter I'm not sure we need all the machinery of Time (as here)? Not really relevant for this PR, except in that a programming error led to a very confusing error message, not at all the intent of using Time and Quantity objects.

Reply:
Actually I think this might be the result of a bare `except:` that makes any error during property access look like the property doesn't exist. Possibly here: https://github.com/nanograv/PINT/blob/6854a9ecb5cc61b7836d3a54b3e9e22691bce14c/src/pint/models/timing_model.py#L223

Reply:
I think the problem is the `.to_value(u.day)` bit.  That doesn't seem to work with even regular Time objects.

However, ```mm.TASC.quantity.to_value('mjd_long')``` works.  But it gives identical results to .value (both are numpy float128), so is there really an advantage?

```
In[31]: mm.TASC.quantity.to_value('mjd_long') == mm.TASC.value                                          
Out[31]: True
```

Reply:
The problem was never about the numbers but about the assumption that a Quantity is in the units you think it's in. Using `x.value` will get you a number that abruptly changes if someone used `u.s` instead of `u.d`. That doesn't really arise in the same way for Time objects, and `MJDParameter` might be documented so that `.value` is always the same as `.to_value('mjd_long')`. (I think `t.mjd_long` is a shorter way to spell `t.to_value('mjd_long')` but both might be only in post-4 Astropy I know I added a number of convenience Time functions around then.)

Reply:
So I just checked, and while you can reset the units of an MJDParameter to be (for instance) `u.s`, if you use `.value` or `.quantity` on it, you still get an MJD.  I think that is because of the "pulsar_mjd" format of the Time instance.  So I think that using `.value` is identical to using `.quantity.to_value('mjd_long')`.  And you are right that (at least with my astropy), that `.mjd_long` doesn't work yet.